cwltool/builder.py

```diff
@@ -604,6 +606,9 @@ def tostr(self, value: Union[MutableMapping[str, str], Any]) -> str:
                     '{} object missing "path": {}'.format(value["class"], value)
                 )
             return value["path"]
+        elif isinstance(value, ScalarFloat):
+            rep = RoundTripRepresenter()
+            return rep.represent_scalar_float(value).value
```

Comment:
Note: if the number was originally expressed in scientific notation, this doesn't convert to decimal (non-scientific) notation.